phys2denoise/tests/test_tasks_integration.py

```diff
@@ -0,0 +1,116 @@
+"""Tests for phys2denoise.tasks and their integration."""
+
+import logging
+import os
+
+import numpy as np
+from loguru import logger
+from physutils import physio
+from physutils.tasks import generate_physio
+from pydra import Submitter, Workflow
+
+import phys2denoise.tasks as tasks
+
+from .utils import create_fake_phys
+
+LGR = logging.getLogger(__name__)
+LGR.setLevel(logging.DEBUG)
+
+
+def test_integration(fake_phys):
+    """Test the integration of phys2denoise tasks."""
+    file_dir = os.path.dirname(os.path.abspath(__file__))
+    export_dir = os.path.join(file_dir, "test_output_data/integration")
+    create_fake_phys()
+
+    physio_file = os.path.abspath("phys2denoise/tests/data/fake_phys.phys")
+
+    wf = Workflow(
+        name="metrics_wf",
+        input_spec=["phys", "fs"],
+        phys=physio_file,
+    )
+    wf.add(
+        generate_physio(
+            name="generate_physio",
+            input_file=wf.lzin.phys,
+            fs=wf.lzin.fs,
+            mode="physio",
+        )
+    )
+    wf.add(
+        tasks.compute_metrics(
+            name="compute_metrics",
+            phys=wf.generate_physio.lzout.out,
+            metrics=[
+                "respiratory_variance",
+                "respiratory_variance_time",
+                "cardiac_phase",
+            ],
+            args={
+                "cardiac_phase": {
+                    "slice_timings": np.linspace(0, 1, 22)[1:-1],
+                    "n_scans": 200,
+                    "t_r": 1.0,
+                },
+            },
+        )
+    )
+    wf.add(
+        tasks.export_metrics(
+            name="export_metrics",
+            phys=wf.compute_metrics.lzout.out,
+            metrics="all",
+            outdir=export_dir,
+            tr=1.0,
+        )
+    )
+    wf.set_output([("result", wf.compute_metrics.lzout.out)])
+
+    with Submitter(plugin="cf") as sub:
+        sub(wf)
```

Comment:
The test is failing here. I'm getting the following error:
`File "/home/user/Documents/physio/phys2denoise/phys2denoise/metrics/chest_belt.py", line 288, in respiratory_variance
    data = physio.check_physio(data, ensure_fs=True, copy=True)
  File "/home/user/Documents/physio/phys2denoise/env/lib/python3.9/site-packages/physutils/physio.py", line 149, in check_physio
    if ensure_fs and np.isnan(data.fs):
TypeError: ufunc 'isnan' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''`

I've tried to investigate a bit more, and turn out that the value of `data.fs` when it's crashing is `NOTHING`:
```
(Pdb) data
Physio(size=18750, fs=_Nothing.NOTHING)
(Pdb) data.fs
NOTHING
(Pdb) type(data.fs)
<enum '_Nothing'>
```

Reply:
I remember this issue, however afaik it was fixed. Can you try replacing the `isnan` check with `pandas.isna()`?

Reply:
Thank you @maestroque for your comment ! `pandas.isna()` is indeed able to evaluate the `NOTHING` value rather than just throwing an error. However, `pandas.isna(NOTHING)` returns `False` so the [value error](https://github.com/physiopy/physutils/blob/master/physutils/physio.py#L149) is not raised.. That causes other problem afterward, for example when [instantiating the Physio object](https://github.com/physiopy/physutils/blob/master/physutils/physio.py#L303C8-L304C34):
```
File "/home/user/Documents/physio/phys2denoise/env/lib/python3.9/site-packages/physutils/physio.py", line 305, in __init__
    self._fs = np.float64(fs)
TypeError: float() argument must be a string or a number, not '_Nothing'
```

Reply:
Follow up question: Are we expecting the value of `data.fs` to be `NOTHING`. Just wondering if the problem might comes from something before e.g. `Workflow` or `generate_physio` are called.

Reply:
The error seems to come from `generate_physio` in `physutils`. Specifically, in that function, when [`load_physio`](https://github.com/physiopy/physutils/blob/master/physutils/tasks.py#L57) is called, it specifies `fs=fs`, which is unecessary and causes issue.

Reply:
All tests are passing with the changes made in [physutils PR #11](https://github.com/physiopy/physutils/pull/11)